src/components/Checkbox/index.jsx

```diff
@@ -1,75 +1,65 @@
-import React from 'react';
-import { v4 } from 'uuid';
+import React, { useMemo } from 'react';
 import PropTypes from 'prop-types';
+import { CustomCheckboxContainer, CustomCheckboxInput } from '@reach/checkbox';
 import cc from 'classcat';
 
 import CheckEmptyIcon from 'lib/icons/react/CheckEmptyIcon';
 import CheckFilledIcon from 'lib/icons/react/CheckFilledIcon';
 import CheckIndeterminateIcon from 'lib/icons/react/CheckIndeterminateIcon';
 
-/**
- * @param {Object} props react props
- * @returns {ReactElement}
- */
-const Checkbox = ({
-  showLabel = true,
-  indeterminate = false,
-  disabled = false,
-  label,
-  inputRef,
-  id = v4(),
-  ...rest
-}) => {
-  const IconUnchecked = CheckEmptyIcon;
-  const IconChecked = indeterminate ? CheckIndeterminateIcon : CheckFilledIcon;
+const Checkbox = ({ checked, label, disabled, showLabel = true, ...rest }) => {
+  const CheckboxIcon = useMemo(() => {
+    if (checked) {
+      if (checked === 'mixed') {
+        return CheckIndeterminateIcon;
+      }
+      return CheckFilledIcon; // true
+    }
+    return CheckEmptyIcon; // false
+  }, [checked]);
+
+  const Element = showLabel ? 'label' : 'div';
 
   return (
-    <div
-      className={cc([{ 'fdsCheckable--disabled': disabled }, 'fdsCheckable fdsCheckbox'])}
-    >
-      <input
-        {...rest}
-        type="checkbox"
-        id={id}
-        className="media--xs"
+    <Element className="alignChild--left--center">
+      <CustomCheckboxContainer
+        checked={checked}
         disabled={disabled}
-        ref={inputRef}
-        aria-label={showLabel ? undefined : label}
-      />
-      <label className="flush--bottom" htmlFor={id}>
-        <span className="fdsCheckable-icon--checked">
-          <IconChecked size="xs" />
-        </span>
-        <span className="fdsCheckable-icon--unchecked">
-          <IconUnchecked size="xs" />
+        className={cc([
+          { 'fdsCheckbox--checked': checked, 'fdsCheckbox--disabled': disabled },
+          'fdsCheckbox-container',
+        ])}
+        {...rest}
+      >
+        <CustomCheckboxInput
+          disabled={disabled}
+          checked={checked}
+          aria-label={!showLabel ? label : undefined}
+          {...rest}
+        />
+        <span aria-hidden className="fdsCheckbox-span">
+          <CheckboxIcon size="xs" />
         </span>
-        {showLabel && <span className="fdsCheckable-label">{label}</span>}
-      </label>
-    </div>
+      </CustomCheckboxContainer>
+      {showLabel && <span className="fdsCheckbox-label">{label}</span>}
+    </Element>
   );
 };
 
 Checkbox.propTypes = {
```

Comment:
I know we spread props, but do you think it's worth explicitly declaring an `onChange` function prop here?